internal/local/structure_v2.go

```diff
@@ -44,8 +44,9 @@ type AuthStructure struct {
 
 // DataSourceStructure represents the v2 Realm app data source structure
 type DataSourceStructure struct {
-	Config map[string]interface{}   `json:"config,omitempty"`
-	Rules  []map[string]interface{} `json:"rules,omitempty"`
+	Config      map[string]interface{}   `json:"config,omitempty"`
+	DefaultRule map[string]interface{}   `json:"default_rule,omitempty"`
```

Comment:
[question] I'm guessing it's currently a moot point, but perhaps we want to add `DefaultRule` to `HTTPServiceStructure` and its corresponding parse/write methods as well?

basically, in switch from structure v1 => v2, we split "services" into "services+http services+data sources"

if "default rule" is a concept on our core `ServiceDesc` interface in the backend, we may just want to future-proof ourselves and match that design by including it here as well.  it seems that `master` on `baas` is already equipped to handle the same based on your recent changes (b/c data sources and http services all just embed the core `Service` type)

Reply:
Yeah, I don't think there's any value in doing this at the moment given that for all non-mongo services we return a nil value for the default rule and I don't think that's liable to change. 

It's unfortunate that the DefaultRule is a concept on the core `ServiceDesc` but that's a tradeoff we had to make for ease of implementation. Once 3rd party services are deprecated in the future, we may be able to revisit this and have separate implementations for Mongo services and have the DefaultRule be exclusive to that interface.